docs/sender_consumers.adoc

```diff
@@ -15,45 +18,38 @@ auto started = async::start_detached(sndr);
 ----
 
 If a sender starts detached, its operation state has to be allocated somewhere.
-That is achieved through an allocator. Given a class T, an allocator supports two
-operations:
+That is achieved through an xref:attributes.adoc#_allocator[allocator]
+determined from the sender's attributes.
+
+To hook into the static xref:attributes.adoc#_allocator[allocation strategy], a
+template argument (representing the name of the allocation domain) can be given
+to `start_detached`.
 
 [source,cpp]
 ----
-template <typename T>
-struct allocator {
-  // allocate space for a T and construct it with Args
-  template <typename... Args> auto construct(Args &&...) -> T *;
-  // destroy and deallocate
-  auto destruct(T const *) -> void;
-};
+auto result = async::start_detached<struct Name>(s);
 ----
 
-The variable templates `async::alloc` and `async::allocation_limit` can be
-specialized to control how detached operation states are handled:
-
-[source,cpp]
-----
-template <typename Name>
-constexpr inline auto allocation_limit = std::size_t{1};
+The default template argument results in a different `static_allocator` for each
+call site, with a default allocation limit of 1. If the allocator's `construct` method
+returns a `nullptr` (presumably because the allocation limit has been reached),
+the result of `start_detached` is an empty optional.
 
-template <typename Name, typename T, std::size_t N>
-inline auto alloc = static_allocator<T, N>{};
-----
+=== `start_detached_unstoppable`
 
-To hook into these allocation strategies, a template argument (representing the
-name of the allocation domain) can be given to `start_detached`.
+`start_detached_unstoppable` behaves identically to `start_detached`, except
+that the return value contains a pointer to a `never_stop_source`, which has the
```

Comment:
question: Why not return a `std::optional` for consistency? If it fails to start does it return a `nullptr` in the current form?

Looking at the test it looks like it returns and optional<pointer-to-a-never-stop-source>

Reply:
Ah yes, perhaps that was not clear. The return value is still an optional. If allocating the operation state fails in either the regular or unstoppable case, an empty optional is returned.

Reply:
Note:
-  we return an optional pointer (for semantic reasons discussed in other conversations)
-  we're using stdx::optional
- stdx::optional<pointer> uses nullptr as a tombstone value